### ISSUE TITLE
ES_JAVA_HOME tests fail on OEL-6, centos-6

### ISSUE THREAD

**Issue by pugnascotia** [elastic/elasticsearch#69271]
**Build scan**:

https://gradle-enterprise.elastic.co/s/nkekxha2jzg2e

**Repro line**:

    ./gradlew :qa:os:destructiveDistroTest.default-rpm

**Reproduces locally?**:

I reproduced this in Vagrant:

```shell
vagrant up oel-6
vagrant ssh oel-6
sudo bash
cd /usr/lib/jvm
wget https://github.com/AdoptOpenJDK/openjdk15-binaries/releases/download/jdk-15.0.2%2B7/OpenJDK15U-jdk_x64_linux_hotspot_15.0.2_7.tar.gz
tar xf OpenJDK15U-jdk_x64_linux_hotspot_15.0.2_7.tar.gz
cd /elasticsearch
export JAVA_HOME=/usr/lib/jvm//usr/lib/jvm/jdk-15.0.2+7
export SYSTEM_JAVA_HOME=$JAVA_HOME
./gradlew :qa:os:destructiveDistroTest.default-rpm
```

I can also reproduce in Vagrant:

* Manually install the RPM
* Move the bundled JDK out of the way:
    ```shell
    mv /usr/share/elasticsearch/jdk /usr/share/elasticsearch/jdk.relocated
    ```
* Specify `ES_JAVA_HOME`:
    ```shell
    echo "ES_JAVA_HOME=/usr/lib/jvm//usr/lib/jvm/jdk-15.0.2+7" >> /etc/sysconfig/elasticsearch
    ```
* Run the service - it will fail immediately
    ```shell
    service elasticsearch start
    ```
* Edit `/etc/sysconfig/elasticsearch` to `s/ES_JAVA_HOME/JAVA_HOME/` - the service will now start

**Applicable branches**:

`7.12`

**Failure history**:

**Failure excerpt**:

```
Startup command should succeed. Stderr: [exitCode = [1] stdout = [Starting elasticsearch: [FAILED]] stderr = [could not find java in bundled JDK at /usr/share/elasticsearch/jdk/bin/java]]
```

**Comment by jasontedor** [comment]
Thanks @pugnascotia. I opened #69278.

**Comment by jasontedor** [comment]
Closed via #69278.
